internal/codespaces/api/api_test.go

```diff
@@ -115,6 +115,145 @@ func TestListCodespaces_unlimited(t *testing.T) {
+			t.Fatalf("expected q = '%s', got '%s", expectedQ, q)
+		}
+
+		if r.URL.Query().Get("per_page") == "" {
+			t.Fatal("Missing 'per_page' param")
+		}
+
+		per_page, _ := strconv.Atoi(r.URL.Query().Get("per_page"))
+		expectedPerPage := 7 // This is currently hardcoded in the API logic
+		if per_page != expectedPerPage {
+			t.Fatalf("expected per_page = %d, got %d", expectedPerPage, per_page)
+		}
+
+		if r.URL.Query().Get("sort") == "" {
+			t.Fatal("Missing 'sort' param")
+		}
+
+		sort := r.URL.Query().Get("sort")
+		expectedSort := "stars" // This is currently hardcoded in the API logic
+		if sort != expectedSort {
+			t.Fatalf("expected sort = '%s', got '%s'", expectedSort, sort)
+		}
+
+		response := struct {
+			Items []*Repository `json:"items"`
+		}{
+			responseRepos,
+		}
+
+		data, _ := json.Marshal(response)
+		fmt.Fprint(w, string(data))
+	}))
+}
+
+func TestGetRepoSuggestions_noSlash(t *testing.T) {
+	searchText := "text"
+	repos := []*Repository{
+		{FullName: "repo1"},
+		{FullName: "repo2"},
+	}
+
+	svr := createFakeSearchReposServer(t, searchText, repos)
+	defer svr.Close()
+
+	api := API{
+		githubAPI: svr.URL,
+		client:    &http.Client{},
+	}
+	ctx := context.TODO()
+	repoNames, err := api.GetCodespaceRepoSuggestions(ctx, searchText)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(repoNames) != len(repos) {
+		t.Fatalf("expected %d repos, got %d", len(repos), len(repoNames))
+	}
+
+	for i, expectedRepo := range repos {
+		expectedName := expectedRepo.FullName
+		actualName := repoNames[i]
+
+		if expectedName != actualName {
+			t.Fatalf("expected repo name for index %d to be %s, got %s", i, expectedName, actualName)
+		}
+	}
+}
+
+func TestGetRepoSuggestions_withSlash(t *testing.T) {
+	searchText := "org/repo"
+	queryText := "repo user:org"
+
+	repos := []*Repository{
+		{FullName: "repo1"},
+		{FullName: "repo2"},
+	}
+
+	svr := createFakeSearchReposServer(t, queryText, repos)
+	defer svr.Close()
+
+	api := API{
+		githubAPI: svr.URL,
+		client:    &http.Client{},
+	}
+	ctx := context.TODO()
+	repoNames, err := api.GetCodespaceRepoSuggestions(ctx, searchText)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(repoNames) != len(repos) {
+		t.Fatalf("expected %d repos, got %d", len(repos), len(repoNames))
+	}
+
+	for i, expectedRepo := range repos {
+		expectedName := expectedRepo.FullName
+		actualName := repoNames[i]
+
+		if expectedName != actualName {
+			t.Fatalf("expected repo name for index %d to be %s, got %s", i, expectedName, actualName)
+		}
+	}
+}
+
+func TestGetRepoSuggestions_badFormat(t *testing.T) {
+	searchText := "org/repo/bad"
+
+	svr := createFakeSearchReposServer(t, "", nil)
+	defer svr.Close()
+
+	api := API{
+		githubAPI: svr.URL,
+		client:    &http.Client{},
+	}
+	ctx := context.TODO()
+	_, err := api.GetCodespaceRepoSuggestions(ctx, searchText)
+	if err == nil {
+		t.Fatal("Expected error")
+	}
+
+	expected := "search value is an incorrect format"
+	if err.Error() != expected {
```

Comment:
Tip: using Sprint to combine the nilness check and the error string check:

```
want := "search value is an incorrect format"
got := fmt.Sprint(err) // (this may be "<nil>")
if got != want {
   t.Fatalf("GetCodespaceRepoSuggestions error was %s, want %s", got, want)
}
```